CRM/Event/BAO/Participant.php

```diff
@@ -836,6 +836,116 @@ public static function deleteParticipant($id) {
     return $participant;
   }
 
+  /**
+   * Retrieves existing participants.
+   *
+   * @param int $contactId
+   *   The contact ID of participants to find.
+   * @param int $eventId
+   *   The event ID of participants to find.
+   * @param bool $onlyCounted
+   *   Whether to only consider registrations with a status with "is_counted".
+   * @param bool $includeOnWaitlist
+   *   Whether to consider registrations with status "On waitlist" when restricing to "is_counted".
+   * @param array $excludeStatus
+   *   A list of registration status to not consider (e.g. for ignoring cancelled registrations).
+   * @param array $filterRoleIds
+   *   A list of participant role IDs to filter for. Registrations with other roles will not be considered.
+   * @param bool $includeTest
+   *   Whether to include test participants.
+   *
+   * @return array<int>
+   *   An array of participant IDs matching the given criteria.
+   * @throws \CRM_Core_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
+  public static function findExistingParticipants(
+    int $contactId,
+    int $eventId,
+    bool $onlyCounted = TRUE,
+    bool $includeOnWaitlist = TRUE,
+    array $excludeStatus = ['Cancelled'],
+    array $filterRoleIds = [],
+    bool $includeTest = FALSE
+  ) {
+    $query = \Civi\Api4\Participant::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('contact_id', '=', $contactId)
+      ->addWhere('event_id', '=', $eventId);
+
+    if ($onlyCounted) {
+      $query
+        ->addJoin('ParticipantStatusType AS participant_status_type', 'LEFT');
```

Comment:
Could be an `INNER` join for performance.

Reply:
You could test that but I doubt it would make a performance difference